### ISSUE TITLE
Project still active?

### ISSUE THREAD

**Issue by mschubert** [awesomebytes/gentoo_prefix_ci_test#8]
Hi,

Great work on the Gentoo Prefix CI, I felt the reliability of the bootstrap improved quite a bit after this was started :+1: (at least that's the impression I had)

I realized that the last full CI run was in April 2021. Is the project still active?

I'd be happy to help reinstate the build process if you're short on time or busy due to other commitments (but I would need some pointers).

**Comment by awesomebytes** [comment]
Hello!

I haven't been using the project for a long time, so I did not even notice!

I took a look at the logs and the reason it's not building is quite simple:
```bash
##[warning]An image label with the label Ubuntu 16.04 does not exist.
```
The Ubuntu 16.04 image used for CI does not exist anymore!

I'll try to revive it now

**Comment by awesomebytes** [comment]
Yep, seems like the Ubuntu 16.04 image got deprecated (as Ubuntu stopped giving security updates).

I changed it to Ubuntu 20.04 and It seems to be running: https://dev.azure.com/12719821/12719821/_build/results?buildId=4735&view=logs&j=d1556f11-2176-5861-bb6c-85d21ddfd4af

I'll check later if it succeeds.

Thanks for the interest @mschubert !

**Comment by awesomebytes** [comment]
The 32 bits version of the project failed, unfortunately:
https://dev.azure.com/12719821/12719821/_build/results?buildId=4742&view=logs&j=93ee70da-8c2e-5470-d3e2-6f2a6c3c7e50&t=1c3cf894-df3d-5e1a-f2c4-8ab33e9ed25f

```
checking for a BSD-compatible install... /usr/bin/install -c
checking whether ln works... yes
checking whether ln -s works... yes
checking for a sed that does not truncate output... /tmp/gentoo/tmp/usr/bin/sed
checking for gawk... gawk
checking for i686-pc-linux-gnu-gcc... i686-pc-linux-gnu-gcc
checking whether the C compiler works... no
configure: error: in `/tmp/gentoo/var/tmp/portage/sys-devel/binutils-2.38-r2/work/build':
configure: error: C compiler cannot create executables
See `config.log' for more details
 * ERROR: sys-devel/binutils-2.38-r2::gentoo failed (configure phase):
 *   (no error message)
 * 
 * Call stack:
 *     ebuild.sh, line 127:  Called src_configure
 *   environment, line 1888:  Called die
 * The specific snippet of code:
 *       "${S}"/configure "${myconf[@]}" || die;
 * 
 * If you need support, post the output of `emerge --info '=sys-devel/binutils-2.38-r2::gentoo'`,
 * the complete build log and the output of `emerge -pqv '=sys-devel/binutils-2.38-r2::gentoo'`.
 * The complete build log is located at '/tmp/gentoo/var/tmp/portage/sys-devel/binutils-2.38-r2/temp/build.log'.
 * The ebuild environment file is located at '/tmp/gentoo/var/tmp/portage/sys-devel/binutils-2.38-r2/temp/environment'.
 * Working directory: '/tmp/gentoo/var/tmp/portage/sys-devel/binutils-2.38-r2/work/build'
 * S: '/tmp/gentoo/var/tmp/portage/sys-devel/binutils-2.38-r2/work/binutils-2.38'
```

This one is still going, though.

**Comment by awesomebytes** [comment]
The bootstrap worked!

https://dev.azure.com/12719821/12719821/_build/results?buildId=4739&view=results

However, the packaging step failed on pushing a release to github. oopsie. I'll take a quick look see if I can fix it, but I may not have time (I'm going on holidays in 2 days)

**Comment by awesomebytes** [comment]
Apparently I can't recover access to the azure devops account (it was created using my university email, which then passed to be an alumni email, and something is not working as expected. Azure DevOps blames my uni, my uni does not have IT support for alumni... fun).

So I think I won't be able to fix the automatic releases anytime soon.

The good news is that you can extract yourself from the docker image:
```bash
docker pull awesomebytes/gentoo_prefix_ci_emerge_system:latest
```
The latest bootstrapped gentoo prefix.

You can also generate a 'release' doing:
https://github.com/awesomebytes/gentoo_prefix_ci/blob/master/azure-pipelines.yml#L242

**Comment by mschubert** [comment]
That's great so far, thanks! Just having the CI confirming that the bootstrap is working already adds a lot of value for me. And as you say I can extract the docker image myself.

For the medium term, I'm wondering if it makes sense to move from Azure to GHA: the latter also allows up to [6 hours runtime per job and 35 days per workflow](https://docs.github.com/en/actions/learn-github-actions/usage-limits-billing-and-administration#usage-limits) - so this should be sufficient to run the Prefix bootstrap (and I'd think it makes maintaining easier).

**Comment by awesomebytes** [comment]
I think the best idea is to keep it all in GitHub, indeed.

This project was one of my first contacts with docker and with cloud continuous integration. So I'd be happy to dedicate a few hours to update it to something more modern.

At the time only Azure DevOps allowed for jobs to take long enough (and having enough CPU/memory/disk) to bootstrap stage 3!

I'd be very happy to have collaborators, by the way. I'll be away on holidays for the next 5/6 weeks, though.

**Comment by mschubert** [comment]
Enjoy your holidays!

And let me see if I come up with a working GHA  until you're back :smile:
